# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/xul/nsMenuPopupFrame.h
# Commit: 3a8608560898
# Full Hash: 3a86085608988058ad0774d0878e5aedfea61f49
# Author: stransky <stransky@redhat.com>
# Date: 2022-11-09 09:58:36
# Regressor Bug: 1761877
# File Overlap Count: 1
# Description:
#   Bug 1761877 [Linux] Add isWaylandDragSource and isWaylandPopup attributes to XULPopupElement r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D161127
# ==============================================================================

diff -r 02c859d135b7 -r 3a8608560898 layout/xul/nsMenuPopupFrame.h
--- a/layout/xul/nsMenuPopupFrame.h	Tue Nov 08 20:27:50 2022 +0000
+++ b/layout/xul/nsMenuPopupFrame.h	Tue Nov 08 20:43:17 2022 +0000
@@ -252,6 +252,9 @@
   // Return true if the popup is for a menulist.
   bool IsMenuList();
 
+  bool IsDragSource() const { return mIsDragSource; }
+  void SetIsDragSource(bool aIsDragSource) { mIsDragSource = aIsDragSource; }
+
   static nsIContent* GetTriggerContent(nsMenuPopupFrame* aMenuPopupFrame);
   void ClearTriggerContent() { mTriggerContent = nullptr; }
   void ClearTriggerContentIncludingDocument();
@@ -631,6 +634,10 @@
   // Whether we have a pending `popuppositioned` event.
   bool mPendingPositionedEvent = false;
 
+  // Whether this popup is source of D&D operation. We can't close such
+  // popup on Wayland as it cancel whole D&D operation.
+  bool mIsDragSource = false;
+
   // When POPUPPOSITION_SELECTION is used, this indicates the vertical offset
   // that the original selected item was. This needs to be used in case the
   // popup gets changed so that we can keep the popup at the same vertical